### PR TITLE
Fix application cannot start when the dubbo interface with @RequestMapping

### PR DESCRIPTION
**Pull request by ascrutae**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#1955

### CODE REVIEW

**Comment by wu-sheng** [comment]
CI fails. And please consider this as an optional agent only, add a document to describe what this optional plugin for, please.

File: apm-sniffer/optional-plugins/pom.xml [hunk 1]
```diff
@@ -43,6 +43,7 @@
     <modules>
         <module>optional-spring-plugins</module>
         <module>trace-ignore-plugin</module>
+        <module>dubbo-patch</module>
```
Comment (wu-sheng) [inline apm-sniffer/optional-plugins/pom.xml]: `dubbo-patch` looks like a strange name. Does this plugin try to resolve dubbo-springmvc conflict? If I am right, please name it `dubbo-springmvc-confict-patch`

**Comment by candyleer** [comment]
Instead of `@RequestMapping`,any things that enhance the `dubbo service interface`will cause this problem, such as `@Trace` on method,`@Service` in `optional-spring-plugins` etc  .
so I preferred this plugin will be put in official plugin that makes it is run well in all this situation if this plugin has no side effect to each `dubbo` versions.

**Comment by wu-sheng** [comment]
> so I preferred this plugin will be put in the official plugin that makes it run well in all this situation if this plugin has no side effect to each dubbo versions.

I don't know whether there are side effects or not. Look like shouldn't have. If @ascrutae and @candyleer both want to put it into the default plugin folder, I think we could try. But please keep this in a separated plugin jar file, in case side effect happens, user could remove it.

How about this?

File: apm-sniffer/apm-sdk-plugin/pom.xml [hunk 2]
```diff
@@ -60,7 +60,8 @@
         <module>activemq-5.x-plugin</module>
         <module>elasticsearch-5.x-plugin</module>
         <module>undertow-plugins</module>
-        <module>rabbitmq-5.x-plugin</module>
+	<module>rabbitmq-5.x-plugin</module>
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/pom.xml]: Format looks like not right.

File: apm-sniffer/apm-sdk-plugin/dubbo-conflict-patch/src/main/java/org/apache/skywalking/apm/plugin/dubbo/patch/MakeWrapperInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        Object ret) {
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes,
+        Throwable t) {
+    }
+
+    private static Wrapper makeWrapper(Class<?> c) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/dubbo-conflict-patch/src/main/java/org/apache/skywalking/apm/plugin/dubbo/patch/MakeWrapperInterceptor.java]: Add a comment linking to the original class and method?

File: apm-sniffer/apm-sdk-plugin/dubbo-conflict-patch/src/main/java/org/apache/skywalking/apm/plugin/dubbo/patch/MakeWrapperInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Matcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.StaticMethodsAroundInterceptor;
+
+public class MakeWrapperInterceptor implements StaticMethodsAroundInterceptor {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/dubbo-conflict-patch/src/main/java/org/apache/skywalking/apm/plugin/dubbo/patch/MakeWrapperInterceptor.java]: Add comments about why do you need this interceptor?

**Pull request merged**